client/my-sites/stats/stats-purchase/index.tsx

```diff
@@ -110,6 +111,10 @@ const StatsPurchasePage = ( {
 			if ( options.isCommercial && ! isCommercialOwned ) {
 				return [ SCREEN_PURCHASE, TYPE_COMMERCIAL ];
 			}
+			// If the site is detected as personal
+			else if ( options.isCommercial === false && ! isCommercialOwned ) {
```

Comment:
```suggestion
			else if ( options.isCommercial === false && ! isPWYWOwned && ! isCommercialOwned ) {
```

Do we need to consider if the user has purchased a PWYW product?

Reply:
I think the condition was correct because it checked if the purchase was not commercial. Since PWYW is accessible through the same screen as the free plan, I guess we shouldn't create a special condition for it.

Reply:
My concern is not to emphasize the entry point of `TYPE_PERSONAL` for personal users who have already purchased the PWYW product since they can not buy any more products on the screen, no matter how they are redirected to the purchase page.

Reply:
I understand the concern - but I think perhaps the PWYW product already existing should be handled in a different location, and as a different issue from this PR. 

Eg. Those with existing PWYW product shouldn't be shown the banners or purchase screen anyhow, and potentially that could be handled in a redirect. So I'm not sure that if they somehow got this far in, that it would matter if their site were assumed as personal already.

Reply:
I'm going to ship this as it is, but noting that we can discuss if we need any additional checks for existing PWYW products at other stages of the process.

Reply:
I missed some nice discussions happening here. 

The [code below](https://github.com/Automattic/wp-calypso/pull/81111/files#diff-7d16e517ddc75ad5e9be5f880600692ed50ecb1e8469c20a43f623fa60b30e27R120 ) would default product to commercial if PWYW already owned, which makes sense. And now we'd know whether they are commercial sites or not. Should we show PWYW or Commercial? I'd lean to commercial as well.

IMO it's an edge case - after all, we don't really show the link for user to upgrade when they have PWYW or Commercial. So let's keep it as is for now because it seems we'll need to implement separate screens for different site types now.